yt_dlp/extractor/archiveorg.py

```diff
@@ -945,3 +946,238 @@ def _real_extract(self, url):
+            params = {arg.get('name'): arg.get('value') for arg in stream.get('keys', []) if arg.get('type') == 'param'}
+            m3u8_doc = self._download_wbm_page(max_stream.get('source'), video_id, note='Downloading m3u8', query=params, fatal=False)
+            if m3u8_doc:
+                # M3U8 document is not valid, so it needs to be fixed
+                m3u8_doc_lines = m3u8_doc.splitlines()
+                modified_m3u8_doc_lines = []
+                url_base = max_stream.get('source').rsplit('/', 1)[0]
+                first_segment = None
+                for line in m3u8_doc_lines:
+                    if line.startswith('#'):
+                        modified_m3u8_doc_lines.append(line)
+                    else:
+                        modified_line = f'{self._WAYBACK_BASE_URL}{url_base}/{line}?{urllib.parse.urlencode(params)}'
+                        modified_m3u8_doc_lines.append(modified_line)
+                        if first_segment is None:
+                            first_segment = modified_line
+                modified_m3u8_doc = '\n'.join(modified_m3u8_doc_lines)
+
+                # Segments may not have been archied. See 101870
+                first_segment_req = self._request_webpage(HEADRequest(first_segment), video_id, note='Check first segment availablity', errnote=False, fatal=False)
+                if first_segment_req:
+                    formats, _ = self._parse_m3u8_formats_and_subtitles(modified_m3u8_doc, ext='mp4', video_id=video_id)
+
+        # For parts of the project MP4 files were archived
+        max_video = max(traverse_obj(vod_data, ('videos', 'list'), []), key=lambda v: traverse_obj(v, ('bitrate', 'video')), default=None)
+        if max_video is not None:
+            video_url = self._WAYBACK_BASE_URL + max_video.get('source')
+            video_req = self._request_webpage(HEADRequest(video_url), video_id, note='Check video availablity', errnote=False, fatal=False)
+            if video_req:
+                formats.append({
+                    'url': video_url
+                })
+
+        # Code from NaverBaseIE
+        automatic_captions = {}
+        subtitles = {}
+        for caption in traverse_obj(vod_data, ('captions', 'list'), []):
+            caption_url = caption.get('source')
+            if not caption_url:
+                continue
+            caption_url = self._WAYBACK_BASE_URL + caption_url
+            sub_dict = automatic_captions if caption.get('type') == 'auto' else subtitles
+            lang = caption.get('locale') or join_nonempty('language', 'country', from_dict=caption) or 'und'
+            if caption.get('type') == 'fan':
+                lang += '_fan%d' % next(i for i in itertools.count(1) if f'{lang}_fan{i}' not in sub_dict)
+            sub_dict.setdefault(lang, []).append({
+                'url': caption_url,
+                'name': join_nonempty('label', 'fanName', from_dict=caption, delim=' - '),
+            })
```

Comment:
Why not subclass from it and call the function?

Reply:
I can't call `_extract_video_info` in `NaverBaseIE` directly due to the function making API requests and returning all formats and not just the archived ones. There would be significant work in modifying the returned value to match what was archived.
Potentially the subtitle extraction part could be moved to its own function, but due to only archiving vtt and not ttml subtitles that also seems like more hassle than it's worth.

Reply:
This should work:
```diff
diff --git a/yt_dlp/extractor/naver.py b/yt_dlp/extractor/naver.py
index e2e6e9728..eae4d07fb 100644
--- a/yt_dlp/extractor/naver.py
+++ b/yt_dlp/extractor/naver.py
@@ -21,6 +21,23 @@
 class NaverBaseIE(InfoExtractor):
     _CAPTION_EXT_RE = r'\.(?:ttml|vtt)'

+    @staticmethod  # NB: Used in VLiveWebArchiveIE
+    def process_subtitles(vod_data, process_url):
+        ret = {'subtitles': {}, 'automatic_captions': {}}
+        for caption in traverse_obj(vod_data, ('captions', 'list', ...)):
+            caption_url = caption.get('source')
+            if not caption_url:
+                continue
+            type_ = 'automatic_captions' if caption.get('type') == 'auto' else 'subtitles'
+            lang = caption.get('locale') or join_nonempty('language', 'country', from_dict=caption) or 'und'
+            if caption.get('type') == 'fan':
+                lang += '_fan%d' % next(i for i in itertools.count(1) if f'{lang}_fan{i}' not in ret[type_])
+            ret[type_].setdefault(lang, []).extend({
+                'url': sub_url,
+                'name': join_nonempty('label', 'fanName', from_dict=caption, delim=' - '),
+            } for sub_url in process_url(caption_url))
+        return ret
+
     def _extract_video_info(self, video_id, vid, key):
         video_data = self._download_json(
             'http://play.rmcnmv.naver.com/vod/play/v2.0/' + vid,
@@ -79,34 +96,18 @@ def get_subs(caption_url):
                 ]
             return [caption_url]

-        automatic_captions = {}
-        subtitles = {}
-        for caption in get_list('caption'):
-            caption_url = caption.get('source')
-            if not caption_url:
-                continue
-            sub_dict = automatic_captions if caption.get('type') == 'auto' else subtitles
-            lang = caption.get('locale') or join_nonempty('language', 'country', from_dict=caption) or 'und'
-            if caption.get('type') == 'fan':
-                lang += '_fan%d' % next(i for i in itertools.count(1) if f'{lang}_fan{i}' not in sub_dict)
-            sub_dict.setdefault(lang, []).extend({
-                'url': sub_url,
-                'name': join_nonempty('label', 'fanName', from_dict=caption, delim=' - '),
-            } for sub_url in get_subs(caption_url))
-
         user = meta.get('user', {})

         return {
             'id': video_id,
             'title': title,
             'formats': formats,
-            'subtitles': subtitles,
-            'automatic_captions': automatic_captions,
             'thumbnail': try_get(meta, lambda x: x['cover']['source']),
             'view_count': int_or_none(meta.get('count')),
             'uploader_id': user.get('id'),
             'uploader': user.get('name'),
             'uploader_url': user.get('url'),
+            **self.process_subtitles(video_data, get_subs),
         }
```
```py
        **NaverBaseIE.process_subtitles(vod_data, lambda x: [self._WAYBACK_BASE_URL + x])
```

Reply:
This looks great. Thank you!